examples/add_person.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/protobuf/examples/tutorial"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+func promptForAddress(r io.Reader) (*tutorial.Person, error) {
+	rd := bufio.NewReader(r)
+	p := &tutorial.Person{}
+	fmt.Print("Enter person ID number: ")
+	if _, err := fmt.Fscan(rd, &p.Id); err != nil {
+		return p, err
+	}
+	if _, err := rd.ReadString('\n'); err != nil {
+		return p, err
+	}
+
+	fmt.Print("Enter name: ")
+	name, err := rd.ReadString('\n')
+	if err != nil {
+		return p, err
+	}
+	p.Name = strings.TrimSpace(name)
+
+	fmt.Print("Enter email address (blank for none): ")
+	email, err := rd.ReadString('\n')
+	if err != nil {
+		return p, err
+	}
+	p.Email = strings.TrimSpace(email)
+
+	for {
+		fmt.Print("Enter a phone number (or leave blank to finish): ")
+		phone, err := rd.ReadString('\n')
+		if err != nil {
+			return p, err
+		}
+		phone = strings.TrimSpace(phone)
+		if phone == "" {
+			break
+		}
+		pn := &tutorial.Person_PhoneNumber{}
+		pn.Number = phone
+
+		fmt.Print("Is this a mobile, home, or work phone? ")
+		ptype, err := rd.ReadString('\n')
+		if err != nil {
+			return p, err
+		}
+		ptype = strings.TrimSpace(ptype)
+		switch ptype {
+		case "mobile":
+			pn.Type = tutorial.Person_MOBILE
+		case "home":
+			pn.Type = tutorial.Person_HOME
+		case "work":
+			pn.Type = tutorial.Person_WORK
+		default:
+			fmt.Println("Unknown phone type.  Using default.")
```

Comment:
it's good practice to include the bad input in the error. Here, use %q to include `ptype`.